spec/details/asserts.lua

```diff
@@ -0,0 +1,823 @@
+local cjson = require("cjson.safe")
```

Comment:
the naming of `spec/details` feels bit confusing. we might want to call something like `spec/tools` or `spec/internal`. not blocker for this specific PR.

Reply:
We will do it in the future, thanks.

Reply:
I agree that `details` is not a good name.

Reply:
I will rename this directory after these PRs are merged.